pkg/aws/eni/node.go

```diff
@@ -624,3 +650,68 @@ func (n *Node) GetMinimumAllocatableIPv4() int {
 
 	return math.IntMin(minimum, (limits.Adapters-index)*maxPerInterface)
 }
+
+// IsPrefixDelegated indicates whether prefix delegation can be enabled on a node.
+// Currently mixed usage of secondary IPs and prefixes is not supported.
+func (n *Node) IsPrefixDelegated() bool {
+	if !n.node.IsPrefixDelegationEnabled() {
+		return false
+	}
+	// Verify if this node is nitro based
+	limits, limitsAvailable := n.getLimitsLocked()
+	if !limitsAvailable {
+		return false
+	}
+	// Allocating prefixes is supported only on nitro instances
+	if limits.HypervisorType != "nitro" {
+		return false
+	}
+	// Verify if all interfaces are prefix delegated. We don't want to enable prefix delegation on nodes that already
+	// use secondary IPs.
+	for _, eni := range n.enis {
+		if len(eni.Addresses) == 0 {
+			continue
+		}
+		if len(eni.Prefixes) == 0 && len(eni.Addresses) > 0 {
+			return false
+		}
+	}
+	return true
+}
+
+// isIPPrefixBased checks if the given IP belongs to any of thr prefixes currently assigned to the node.
```

Comment:
typo

Reply:
This method doesn't exist anymore after incorporating feedback from https://github.com/cilium/cilium/pull/18463#discussion_r818861043